src/Aquifer.Jobs/Services/TranslationProcessors/FrenchTranslationProcessingService.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Text.RegularExpressions;
+using Aquifer.Common.Utilities;
+
+namespace Aquifer.Jobs.Services.TranslationProcessors;
+
+public sealed partial class FrenchTranslationProcessingService : ILanguageSpecificTranslationProcessingService
+{
+    private const char NarrowNonBreakingSpace = '\u202f';
+    private const char NonBreakingSpace = '\u00a0';
+    private const string HtmlEncodedNonBreakingSpace = "&nbsp;";
+
+    private static readonly IReadOnlySet<string> s_RegexItemsThatHaveNnbspAfterSelf = new HashSet<string>
+    {
+        @"\bv\.", // v.
+        @"\bp\.", // p.
+        "«",
+        "=",
+        @"\bapr\.", // apr.
+        @"\bav\." // av.
+    };
+
+    private static readonly IReadOnlySet<string> s_RegexItemsThatHaveNnbspBeforeSelf = new HashSet<string>
+    {
+        "»",
+        "!",
+        @"\?",
+        ":",
+        ";"
+    };
+
+    private readonly string _nnbspAfterPattern = @$"({string.Join("|", s_RegexItemsThatHaveNnbspAfterSelf)})(?=[A-Za-zà-üÀ-Ü0-9\x20])";
+    private readonly string _nnbspBeforePattern = @$"(\b|\x20)({string.Join("|", s_RegexItemsThatHaveNnbspBeforeSelf)})";
+
+    public string Iso6393Code => "FRA";
+
+    public async Task<string> PostProcessHtmlAsync(string html, CancellationToken cancellationToken)
+    {
+        var processedHtml = await HtmlUtilities.ProcessHtmlTextContentAsync(html,
+            async text =>
+            {
+                text = await SetNarrowNonBreakingSpacesAsync(text);
+                text = await SetCorrectBibleBookAbbreviation(text);
+
+                return text;
+            });
+        return processedHtml;
+    }
+
+    private Task<string> SetNarrowNonBreakingSpacesAsync(string text)
+    {
+        text = text.Replace(NonBreakingSpace, NarrowNonBreakingSpace).Replace(HtmlEncodedNonBreakingSpace, $"{NarrowNonBreakingSpace}");
+
+        text = Regex.Replace(text, _nnbspAfterPattern, m => m.Value + NarrowNonBreakingSpace);
+        text = Regex.Replace(text, _nnbspBeforePattern, m => NarrowNonBreakingSpace + m.Value);
+
+        // The regex matches the value but not the character before/after, so if a space comes before/after we'll end up with two spaces.
+        text = text.Replace($"{NarrowNonBreakingSpace} ", $"{NarrowNonBreakingSpace}")
+            .Replace($" {NarrowNonBreakingSpace}", $"{NarrowNonBreakingSpace}")
+            // Replace this dash specifically with non-breaking dash
+            .Replace("J.-C.", "J.‑C.");
+
+        return Task.FromResult(text);
+    }
+
+    private Task<string> SetCorrectBibleBookAbbreviation(string text)
+    {
+        text = BibleBookRegex()
+            .Replace(text,
+                m => m.Value.ToLower().Replace(NarrowNonBreakingSpace, ' ') switch
```

Comment:
I generally suggest a `Dictionary<string, string>` for something like this though pattern matching does make it concise at least.  I believe `switch` pattern matching will ultimately write all of this out in the IL as conditional `if`/`else`/`elseif` logic whereas a `Dictionary` would use `O(N)` lookup time of a data structure.